### PR TITLE
#3991 Add navigation library to mobile

### PR DESCRIPTION
**Pull request by desfero**
**Technical notes**
This PR connects wallet with `react-navigation` library with some example routing. 
Please check the comment in a ticket with brief summary of why `react-navigation` was finally chosen as a library to go.

### CODE REVIEW

File: packages/wallet/package.json [hunk 1]
```diff
@@ -13,8 +13,15 @@
   },
   "dependencies": {
     "@neufund/shared": "^1.0.0",
-    "react": "^16.12.0",
+    "@react-native-community/masked-view": "^0.1.6",
+    "@react-navigation/native": "^5.0.1",
+    "@react-navigation/stack": "^5.0.1",
+    "react": "16.9.0",
```
Comment (desfero) [inline packages/wallet/package.json]: `react` was moved back to the version compatible with current `react-native` version

**Comment by DmitriyNoa** [comment]
Tested on IOS and Android emulators. Works as expected.

File: packages/wallet/app/appRoutes.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const appRoutes = {
+  Landing: "Landing",
```
Comment (DmitriyNoa) [inline packages/wallet/app/appRoutes.ts]: Maybe let's keep constants names capitalized as we have in in web package? E.g. APP_ROUTES, LANDING?

File: packages/wallet/index.js [hunk 3]
```diff
@@ -1,9 +1,7 @@
-/**
```
Comment (DmitriyNoa) [inline packages/wallet/index.js]: Is it possible to move this to TS?

File: packages/wallet/app/appRoutes.ts (hunk 2 above)
Comment (desfero) [reply packages/wallet/app/appRoutes.ts]: It depends where the constants are. I see `appRoutes` on the web is also camelCased, same with `externalRoutes`. Maybe let's, for now, stick with the web convention and then later decide if we want to refactor as it's just simple rename :)

File: packages/wallet/index.js (hunk 3 above)
Comment (desfero) [reply packages/wallet/index.js]: makes sense to keep consistency, fixed

**Pull request merged**